crates/pallet-domains/src/block_tree.rs

```diff
@@ -180,6 +181,16 @@ pub(crate) fn verify_execution_receipt<T: Config>(
         return Ok(());
     }
 
+    // Check if the ER has at least 2 trace root (for Initialization and Finalization of block at least)
+    if execution_trace.len() < 2 {
+        log::error!(
```

Comment:
This log is duplicated, `validate_unsigned` already has a log for invalid bundle.